### PR TITLE
17029 - Help with Starting a Business

### PR DESCRIPTION
**Pull request by ketaki-deodhar**
*Issue #:*
https://github.com/bcgov/entity/issues/17029

*Description of changes:*
Add Help with Starting a Business. 

image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2485-7hwzaw3q.web.app

**Comment by severinbeauvais** [comment]
Your first image above doesn't match the temp URL or the uxpin. Either link in the uxpin, or provide a recent screenshot, or just remove it since reviewers can go look it up in the ticket hopefully.

File: auth-web/src/components/auth/manage-business/EntityManagement.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/EntityManagement.vue:536]: Please fix the import order.

Don't worry about the other lint warnings since you didn't touch those files.

File: auth-web/src/components/auth/common/ExpandableHelp.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/common/ExpandableHelp.vue:53]: You can use a Vuetify class in the template for this property.

File: auth-web/src/components/auth/common/ExpandableHelp.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/common/ExpandableHelp.vue:49]: Please fix the indentation below.

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2485-7hwzaw3q.web.app

**Review by seeker25 — APPROVED** [review]
LGTM - Merge when Severin's comments are resolved + CI is passing.

File: auth-web/src/components/auth/manage-business/EntityManagement.vue
Comment (JazzarKarim) [inline auth-web/src/components/auth/manage-business/EntityManagement.vue:557]: Please Ketaki if you can sort the components here in alphabetical order 🙌

**Review by JazzarKarim — APPROVED** [review]
Aside from Sev's comments, LGTM! 👍

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2485-7hwzaw3q.web.app

File: auth-web/src/components/auth/common/ExpandableHelp.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/common/ExpandableHelp.vue:42]: I see you changed from Composition API to Class API. I don't care either way.

File: auth-web/src/components/auth/common/ExpandableHelp.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/common/ExpandableHelp.vue:45]: Please add "readonly" and "!" as per other props.

File: auth-web/src/components/auth/common/ExpandableHelp.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/common/ExpandableHelp.vue]: I was thinking about ["align-self-center"](https://v2.vuetifyjs.com/en/styles/flex/#flex-align-self) but I guess, in the end, you didn't need it.

File: auth-web/src/components/auth/manage-business/EntityManagement.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/EntityManagement.vue:1152]: Why not simply remove (or adjust) the bottom margin on "view-header"?

File: auth-web/src/components/auth/manage-business/EntityManagement.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/EntityManagement.vue:184]: Looks really clean! But do you need this div?

File: auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue:2]: I think it's good practice for components to have an id on their top element (or a classname if there can be multiple copies of the component).

PS I like the id to match the component name.

File: auth-web/src/components/auth/common/ExpandableHelp.vue
Comment (ketaki-deodhar) [reply auth-web/src/components/auth/common/ExpandableHelp.vue]: Do not need it. Looks pretty close to UX pin after a lot of CSS changes

File: auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue:20]: Add "open-in-new" icon after this link, per the uxpin.

**Review by severinbeauvais — APPROVED** [review]
Just some really small comments but LGTM thereafter.

File: auth-web/src/components/auth/manage-business/EntityManagement.vue
Comment (ketaki-deodhar) [reply auth-web/src/components/auth/manage-business/EntityManagement.vue:1152]: I will see if I can reduce this space with some more CSS adjustment. After removing the bottom margin:
image

Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/EntityManagement.vue:1152]: If I delete the margin-bottom on the H1, and change the margin-bottom to 0.75rem on the view-header, it looks like this for me:

image

Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/EntityManagement.vue:1152]: Why do your screenshots keep showing "Add an existing Business or Name Request" ? That's main branch code, not feature branch code. (But your PR and temp URL look correct.)

Comment (ketaki-deodhar) [reply auth-web/src/components/auth/manage-business/EntityManagement.vue]: Yes that is exactly what I found. Override .view-header margin.

File: auth-web/src/components/auth/manage-business/EntityManagement.vue
Comment (ketaki-deodhar) [reply auth-web/src/components/auth/manage-business/EntityManagement.vue]: removed div

File: auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue:61]: Why did you add all these non-blanking spaces?

File: auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue:164]: This lines is a third size spec you give this icon. (The other is class "small-icon" and "size=16".)

Could you use one of the [v-icon props](https://v2.vuetifyjs.com/en/api/v-icon/#props) to set its size,m or at least remove the 2 specs that don't work?

File: auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue
Comment (ketaki-deodhar) [reply auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue:61]: To adjust the text alignment. Otherwise the link text looked broken and misaligned

File: auth-web/src/components/auth/manage-business/EntityManagement.vue
Comment (ketaki-deodhar) [reply auth-web/src/components/auth/manage-business/EntityManagement.vue]: > > Why do your screenshots keep showing "Add an existing Business or Name Request" ? That's main branch code, not feature branch code. (But your PR and temp URL look correct.)
> 
> I am pretty sure I branched off from feature branch. I will double check why it shows main branch code

I think this is from another PR. Not sure but looks like this PR https://github.com/bcgov/sbc-auth/pull/2469

File: auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue
Comment (ketaki-deodhar) [reply auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue:164]: Using v-icon size. removed class. Missed out on removing this earlier

File: auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue:61]: Wrong solution to the problem. And I can break it (and other sentences) by making my screen narrowing and forcing sentence wrapping.

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2485-7hwzaw3q.web.app

File: auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue]: Do you still need class "small-icon" on line 21?

File: auth-web/src/components/auth/manage-business/EntityManagement.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/EntityManagement.vue]: Oh well, this PR's changes, and the temp URL, look ok.

File: auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue
Comment (ketaki-deodhar) [reply auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue]: You are right. nope do not need it

File: auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue:61]: It's probably related to your li / marker / :before CSS. Maybe that could be simplified?

File: auth-web/src/components/auth/manage-business/EntityManagement.vue
Comment (ketaki-deodhar) [reply auth-web/src/components/auth/manage-business/EntityManagement.vue]: Yes. It is FF so in DEV it looks different than the temp URL.

File: auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue
Comment (ketaki-deodhar) [reply auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue:61]: Okay taking a look at that

File: auth-web/src/components/auth/manage-business/EntityManagement.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/EntityManagement.vue]: So you were taking screenshots from Dev, not from the temp URL or from your local? That would explain it.

File: auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue
Comment (ketaki-deodhar) [inline auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue:152]: Simplified css for list items

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2485-7hwzaw3q.web.app

File: auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/StartNewBusinessHelp.vue]: The new code looks great, and it works perfectly! Nice! Well done 👍

**Review by severinbeauvais — APPROVED** [review]
LGTM

**Review by severinbeauvais — APPROVED** [review]
LGTM. I think you can merge when ready.

**Comment by seeker25** [comment]
Awesome - I'll try to move over the Pinia changes tomorrow to the feature branch after this merged

**Pull request merged**
